src/components/magnet/field-line.tsx

```diff
@@ -21,11 +22,17 @@ export default PixiComponent<IProps, PIXI.Graphics>("FieldLine", {
     return new PIXI.Graphics();
   },
   applyProps: (instance, oldProps, newProps) => {
-    const { x, y, magnets, magnetModels, internal } = newProps;
+    const { x, y, magnets, magnetModels, internal, magnetLength } = newProps;
     instance.clear();
     instance.lineStyle(3, 0xBBBBBB);
     instance.moveTo(x, y);
 
+    if (internal) {
+      // fake a straight line back
+      instance.lineTo(x - magnetLength!, y);
```

Comment:
Having to force `magnetLength` to exist here feels a little off - I'd consider scrapping the `internal` prop and just passing a `fixedInternalLength` prop instead